src/js/apps/patients/sidebar/filters-sidebar_app.js

```diff
@@ -39,8 +52,21 @@ export default App.extend({
 
     this.showChildView('customFilters', customFiltersView);
   },
+  showStatesFiltersView() {
+    const statesFiltersView = new StatesFiltersView({
+      collection: this.defaultStatesFilter,
+      model: this.getState(),
+    });
+
+    this.showChildView('statesFilters', statesFiltersView);
+  },
   onClearFilters() {
-    this.getState().clear();
+    const defaultStates = this.defaultStatesFilter.map('id');
+    const worklistId = this.getState('worklistId');
+
+    this.getState().clear().set({ states: defaultStates });
+
+    if (worklistId) this.setState({ worklistId });
```

Comment:
This is shared by both the worklists and schedule. So this ensures that the `state.filters.worklistId` is only added to worklists. Not to the schedule.

On the schedule, `this.getState('worklistId')` will be `undefined`, so it won't be added to the state.